app/src/androidTest/java/ch/epfl/polychef/fragments/RateRecipeFragmentsHomeTest.java

```diff
@@ -82,31 +97,120 @@ public void createInstanceOfRateRecipeFragmentDoesNotThrowError(){
+        assertEquals((int)images.get(3).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(4).getTag(), ratingBar.getEmptyImageResource());
+    }
 
-        onView(withId(R.id.buttonRate)).perform(NestedScrollViewHelper.nestedScrollTo(),click());
-        onView(withId(R.id.RateChoices)).perform(click());
-        String star= nbStars<2?" star":" stars";
-        onData(allOf(is(instanceOf(String.class)), is(nbStars+star))).perform(click());
-        onView(withId(R.id.RateChoices)).check(matches(withSpinnerText(containsString(nbStars+star))));
+    @Test
+    public void correctImageResourceAreDisplayedWithHalf() throws Throwable {
+        //Not possible to rate half but here we force it to test for other display place of the rating bar
+        runOnUiThread(() -> ratingBar.setRate(2.5));
+        ArrayList<ImageView> images = ratingBar.getAllStarsImage();
+        assertNotNull(images);
+        assertEquals((int)images.get(0).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(1).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(2).getTag(), ratingBar.getHalfImageResource());
+        assertEquals((int)images.get(3).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(4).getTag(), ratingBar.getEmptyImageResource());
+    }
+
+    @Test
+    public void correctImageResourceAreDisplayedWithLessThanZero() throws Throwable {
+        runOnUiThread(() -> ratingBar.setRate(-5));
+        ArrayList<ImageView> images = ratingBar.getAllStarsImage();
+        assertNotNull(images);
+        assertEquals(0, ratingBar.getRate());
+        assertEquals((int)images.get(0).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(1).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(2).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(3).getTag(), ratingBar.getEmptyImageResource());
+        assertEquals((int)images.get(4).getTag(), ratingBar.getEmptyImageResource());
+    }
+
+    @Test
+    public void correctImageResourceAreDisplayedWithMoreThanFive() throws Throwable {
+        runOnUiThread(() -> ratingBar.setRate(10));
+        ArrayList<ImageView> images = ratingBar.getAllStarsImage();
+        assertNotNull(images);
+        assertEquals(5, ratingBar.getRate());
+        assertEquals((int)images.get(0).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(1).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(2).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(3).getTag(), ratingBar.getFullImageResource());
+        assertEquals((int)images.get(4).getTag(), ratingBar.getFullImageResource());
+    }
+
+    private void rateCurrentRecipeNStars(int nbStars){
```

Comment:
You should test that something happens after the click

Reply:
But I actually do in the image resource tests when I test that the correct image has been displayed meaning the click works ?